tngo application structure (#1035)
plotly/dash-docs: dash_docs/chapters/dash_enterprise/application_structure/requirements_conda.md

We recommend including all Python packages and their version numbers inside your 
`conda-requirements.txt`.

The packages and version numbers that are currently used in your development environment
can be inspected by running `conda list` in your terminal.
This will include all of the "subdependencies" of the packages that you rely on.
For example, `dash` itself relies on 17 other packages that have their own version numbers.

If you don't include the subdependencies in your `conda-requirements.txt` file, then Dash Enterprise 
will install the latest versions of these subdependencies when you deploy your app for the first time
or restart the application. This could mean that your deployed application uses different versions
of packages than your development environment which could cause unpredictable errors.

Certain versions packages that you install on Windows may not be available on the Linux containers
that Dash Enterprise uses or vice versa. If you run into cross-operating system package issues,
we recommend using Dash Enterprise Workspaces.

Redeploying the application will not reinstall dependencies unless you have changed 
the `conda-requirements.txt` file.

> This following file was generated by running `conda list --export > conda-requirements.txt` 

Reviewer: When you double check `conda-requirements.txt`, also double check that this command results in the correctly formatted file. conda could list these packages in a different format.

with an environment that used the latest packages available as of December 2020 with 

> Dash Enterprise 4.0.1.

Reviewer: n the paragraph above, we mention:
> This will include all of the "subdependencies" of the packages that you rely on. For example, dash itself relies on 17 other packages that have their own version numbers.

But then in this requirements example we only have 6 packages. I would expect to see a similar set of packages as in `pip`. Can you comment on what you tried here? We might need to run a different conda command.

> Dash Enterprise 4.0.1.

Reviewer: I'll look into this

```
amqp==5.0.2
billiard==3.6.3.0
Brotli==1.0.9
celery==5.0.3
certifi==2020.11.8
cffi==1.14.4
chardet==3.0.4
click==7.1.2
click-didyoumean==0.0.3
click-plugins==1.1.1
click-repl==0.1.6
cryptography==3.2.1
dash==1.17.0
dash-core-components==1.13.0
dash-design-kit==1.5.4
dash-enterprise-auth==0.0.4
dash-html-components==1.1.1
dash-renderer==1.8.3
dash-snapshots==1.3.2
dash-table==4.11.0
Flask==1.1.2
Flask-Compress==1.8.0
Flask-SQLAlchemy==2.4.4
future==0.18.2
gunicorn==20.0.4
idna==2.10
importlib-metadata==3.1.1
itsdangerous==1.1.0
Jinja2==2.11.2
kombu==5.0.2
MarkupSafe==1.1.1
numpy==1.19.4
packaging==20.7
pandas==1.1.4
pg8000==1.16.6
plotly==4.13.0
prompt-toolkit==3.0.8
pycparser==2.20
PyJWT==1.7.1
pyOpenSSL==20.0.0
pyparsing==2.4.7
python-dateutil==2.8.1
pytz==2020.4
redis==3.5.3
requests==2.25.0
retrying==1.3.3
scramp==1.2.0
six==1.15.0
SQLAlchemy==1.3.20
urllib3==1.26.2
vine==5.0.0
wcwidth==0.2.5
Werkzeug==1.0.1
zipp==3.4.0
```

> For packages not available on conda channels — notably all Dash Enterprise packages 

Reviewer: Break this out into a new section with the `# requirements.txt` header.

`conda-requirements.txt`.

The packages and version numbers that are currently used in your development environment
can be inspected by running `conda list` in your terminal.
This will include all of the "subdependencies" of the packages that you rely on.
For example, `dash` itself relies on 17 other packages that have their own version numbers.

If you don't include the subdependencies in your `conda-requirements.txt` file, then Dash Enterprise 
will install the latest versions of these subdependencies when you deploy your app for the first time
or restart the application. This could mean that your deployed application uses different versions
of packages than your development environment which could cause unpredictable errors.

Certain versions packages that you install on Windows may not be available on the Linux containers
that Dash Enterprise uses or vice versa. If you run into cross-operating system package issues,
we recommend using Dash Enterprise Workspaces.

Redeploying the application will not reinstall dependencies unless you have changed 
the `conda-requirements.txt` file.

This following file was generated by running `conda list --export > conda-requirements.txt` 

> with an environment that used the latest packages available as of December 2020 with 
> Dash Enterprise 4.0.1.

Reviewer: ```suggestion
with an environment that installed `dash`, `dash-table`, `dash-core-components`, `dash-html-components`, `pandas` & `gunicorn` in December 2020 with 
```

```
dash-core-components==1.13.0
dash-html-components==1.1.1
dash-table==4.11.0
dash==1.17.0
gunicorn==19.9.0
pandas==0.25.3

> ```

Reviewer: ```
# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: linux-64
_libgcc_mutex=0.1=main
blas=1.0=mkl
ca-certificates=2020.12.8=h06a4308_0
certifi=2020.12.5=py38h06a4308_0
click=7.1.2=py_0
dash=1.18.1=pyhd8ed1ab_0
dash-core-components=1.14.1=pyhd8ed1ab_0
dash-html-components=1.1.1=pyh9f0ad1d_0
dash-renderer=1.8.3=pyhd3eb1b0_0
dash-table=4.11.1=pyhd8ed1ab_0
flask=1.1.2=py_0
flask-compress=1.4.0=py_0
future=0.18.2=py38_1
gunicorn=20.0.4=py38_0
intel-openmp=2020.2=254
itsdangerous=1.1.0=py_0
jinja2=2.11.2=py_0
ld_impl_linux-64=2.33.1=h53a641e_7
libedit=3.1.20191231=h14c3975_1
libffi=3.3=he6710b0_2
libgcc-ng=9.1.0=hdf63c60_0
libstdcxx-ng=9.1.0=hdf63c60_0
markupsafe=1.1.1=py38h7b6447c_0
mkl=2020.2=256
mkl-service=2.3.0=py38he904b0f_0
mkl_fft=1.2.0=py38h23d657b_0
mkl_random=1.1.1=py38h0573a6f_0
ncurses=6.2=he6710b0_1
numpy=1.19.2=py38h54aff64_0
numpy-base=1.19.2=py38hfa32c7d_0
openssl=1.1.1i=h27cfd23_0
pandas=1.1.4=py38h0ef3d22_0
pip=20.3.3=py38h06a4308_0
plotly=4.14.3=py_0
python=3.8.5=h7579374_1
python-dateutil=2.8.1=py_0
python_abi=3.8=1_cp38
pytz=2020.5=pyhd3eb1b0_0
pyyaml=5.3.1=py38h7b6447c_1
readline=8.0=h7b6447c_0
retrying=1.3.3=py_2
setuptools=51.1.2=py38h06a4308_3
six=1.15.0=py38h06a4308_0
sqlite=3.33.0=h62c20be_0
tk=8.6.10=hbc83047_0
werkzeug=1.0.1=py_0
wheel=0.36.2=pyhd3eb1b0_0
xz=5.2.5=h7b6447c_0
yaml=0.2.5=h7b6447c_0
zlib=1.2.11=h7b6447c_3
```

> 

Reviewer: ```suggestion
Note that in `conda`, unlike `pip`, `dash-core-components`, `dash-html-components`, and `dash-table` are not installed automatically when installing `dash`. They must be installed separately.

To see which versions of these packages are released with `dash`, see https://github.com/plotly/dash/blob/v1.18.1/requires-install.txt and change the tag to the version of `dash`.
```

## requirements.txt

For packages not available on conda channels — notably all Dash Enterprise packages — 
list them in a `requirements.txt` file located in your app's root directory.

The `requirements.txt` file may resemble:

```
--extra-index-url=https://<dash-enterprise-server>/Docs/packages
dash-design-kit==1.5.4 
dash-enterprise-auth==0.0.4
dash-snapshots==1.3.2
dash-design-kit==1.5.4

```

> 

Reviewer: ```suggestion
See https://<your-dash-enterprise-server>/Docs/read-this-first for latest version numbers of these packages.
```

(And let's make a similar change in the python `requirements.txt` section too if applicable)
